src/table-theme-colors.js

```diff
@@ -1,8 +1,23 @@
-import { isDarkColor } from './table/utils/color-utils';
+import { isDarkColor, isTransparentColor } from './table/utils/color-utils';
 
 export default function tableThemeColors(theme) {
-  const backgroundColor = theme.getStyle('object', 'straightTable', 'backgroundColor');
-  const isBackgroundDarkColor = isDarkColor(backgroundColor);
+  const qvInnerObject = document.querySelector('.qv-object .qv-inner-object');
+  const objectBackgroundColorFromCSS = qvInnerObject && window.getComputedStyle?.(qvInnerObject).backgroundColor;
+  const qvPanelSheet = document.querySelector('.qv-panel-sheet');
+  const sheetBackgroundColorFromCSS = qvPanelSheet && window.getComputedStyle?.(qvPanelSheet).backgroundColor;
+  const opacityFromCSS = qvInnerObject && window.getComputedStyle?.(qvInnerObject).opacity;
+
+  const tableBackgroundColorFromTheme = theme.getStyle('', '', 'object.straightTable.backgroundColor');
+  const backgroundColorFromTheme = theme.getStyle('object', 'straightTable', 'backgroundColor');
+
+  const backgroundColor = tableBackgroundColorFromTheme || objectBackgroundColorFromCSS || backgroundColorFromTheme;
+  const isBackgroundTransparentColor = isTransparentColor(
+    backgroundColor,
+    tableBackgroundColorFromTheme ?? opacityFromCSS
```

Comment:
this is a bit weird that you would pass `tableBackgroundColorFromTheme` if it exists, it is not opacity. If we would convert to ts, it would not like this.

Reply:
fixed